src/SwedbankPay/Path.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+namespace SwedbankPay;
+
+class Path
+{
+        /**
+    * This function is to replace PHP's extremely buggy realpath().
+    * https://stackoverflow.com/a/4050444/61818
+    *
+    * @param string The original path, can be relative etc.
+    * @return string The resolved path, it might not exist.
+    */
+    public static function resolve($path)
+    {
+        // whether $path is unix or not
+        $unipath = strlen($path) == 0 || $path[0] != '/';
+
+        // attempts to detect if path is relative in which case, add cwd
+        if (strpos($path, ':') === false && $unipath) {
+            $path = getcwd() . DIRECTORY_SEPARATOR . $path;
+        }
+
+        // resolve path parts (single dot, double dot and double delimiters)
+        $path = str_replace(array('/', '\\'), DIRECTORY_SEPARATOR, $path);
+        $parts = array_filter(explode(DIRECTORY_SEPARATOR, $path), 'strlen');
+        $absolutes = array();
+
+        foreach ($parts as $part) {
+            if ('.'  == $part) {
+                continue;
+            }
+
+            if ('..' == $part) {
+                array_pop($absolutes);
+            } else {
+                $absolutes[] = $part;
+            }
+        }
+
+        $path = implode(DIRECTORY_SEPARATOR, $absolutes);
+
+        // resolve any symlinks
+        if (file_exists($path) && linkinfo($path) > 0) {
+            $path = readlink($path);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The use of function readlink() is discouraged](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-sdk-php/pullRequest?prid=5303907)